graspologic/plot/plot.py

```diff
@@ -671,6 +674,248 @@ def pairplot(
+    cluster_palette : dict, optional, default: dictionary using 'Set1'
+    alpha : float, optional, default: 0.7
+        Opacity value of plotter markers between 0 and 1
+    """
+    sns.scatterplot(
+        data=data, x=X[:, j], y=X[:, k], ax=ax, hue="labels", palette=label_palette
+    )
+    for i, (mean, covar) in enumerate(zip(means, covariances)):
+        v, w = linalg.eigh(covar)
+        v = 2.0 * np.sqrt(2.0) * np.sqrt(v)
+        u = w[0] / linalg.norm(w[0])
+        # Plot an ellipse to show the Gaussian component
+        angle = np.arctan(u[1] / u[0])
+        angle = 180.0 * angle / np.pi
+        ell = mpl.patches.Ellipse(
+            [mean[j], mean[k]],
+            v[0],
+            v[1],
+            180.0 + angle,
+            color=cluster_palette[i],
+        )
+        ell.set_clip_box(ax.bbox)
+        ell.set_alpha(alpha)
+        ax.add_artist(ell)
+        ax.set_xticks([]), ax.set_yticks([])
+        ax.set_ylabel(k), ax.set_xlabel(k)
+        ax.legend().remove()
+
+
+def pairplot_with_gmm(
+    X,
+    gmm,
+    labels=None,
+    cluster_palette=None,
+    label_palette=None,
+    title=None,
+    context="talk",
+    font_scale=1,
+    alpha=0.7,
+):
+    r"""
+    Plot pairwise relationships in a dataset.
+    By default, this function will create a grid of Axes such that each
+    dimension in data will by shared in the y-axis across a single row and
+    in the x-axis across a single column.
+
+    Parameters
+    ----------
+    X : np.ndarray (2D)
+        input graph
+    gmm: Gaussian Mixture Model object
+    labels : 1d np.ndarray or list, same length as dimensions of X
+        Labels for different categories of graph nodes
+    label_palette : str or dict, optional,
+        with no input : dictionary using 'Set1'
+    cluster_palette : str or dict, optional, default: dictionary using 'Set1'
+        with no input : dictionary using 'Set1'
+    title : string
+        default : "Fit a Gaussian mixture using " + n_components + " components"
+    context :  None, or one of {talk (default), paper, notebook, poster}
+        Seaborn plotting context
+    font_scale : float, optional, default: 1
+        Separate scaling factor to independently scale the size of the font
+        elements.
+    alpha : float, optional, default: 0.7
+        Opacity value of plotter markers between 0 and 1
+    """
+    # Handle X
+    if not isinstance(X, (list, np.ndarray)):
+        msg = "X must be array-like, not {}.".format(type(X))
+        raise TypeError(msg)
+
+    # Handle labels
+    if labels is not None:
+        if not isinstance(labels, (list, np.ndarray)):
+            msg = "Y must be array-like or list, not {}.".format(type(labels))
+            raise TypeError(msg)
+        elif X.shape[0] != len(labels):
+            msg = "Expected length {}, but got length {} instead for Y.".format(
+                X.shape[0], len(labels)
+            )
+            raise ValueError(msg)
```

Comment:
Y is not the name of a parameter.

Reply:
also may be able to use `check_X_y` from sklearn

Reply:
https://github.com/microsoft/graspologic/pull/541/commits/d909082697eee2a7de85f06fafba4338735d5635